src/mSupplyMobileApp.js

```diff
@@ -83,21 +83,27 @@ class MSupplyMobileAppContainer extends React.Component {
     this.handleBackEvent = this.handleBackEvent.bind(this);
     this.getCanNavigateBack = this.getCanNavigateBack.bind(this);
     this.renderPageTitle = this.renderPageTitle.bind(this);
-    this.scheduler.schedule(this.synchronise,
-                            SYNC_INTERVAL);
+    this.scheduler.schedule(this.synchronise, SYNC_INTERVAL);
     this.scheduler.schedule(() => {
-      if (this.state.currentUser !== null) { // Only reauthenticate if currently logged in
+      if (this.state.currentUser !== null) {
+        // Only reauthenticate if currently logged in
         this.userAuthenticator.reauthenticate(this.onAuthentication);
       }
     }, AUTHENTICATION_INTERVAL);
   }
 
   componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.handleBackEvent);
+    if (this.state.isInitialised && this.synchroniser.isSyncing()) {
```

Comment:
Listeners and sync do not appear to operate when the app is in the background. Perhaps applies to all app code, hence componentWillUnmount doesn't get called when the app is closed while in background.